sdk/Trace/ExporterFactory.php

```diff
@@ -0,0 +1,165 @@
+            return null;
+        }
+
+        $contribName = strtolower($strArr[0]);
+        $endpointUrl = $strArr[1];
+
+        if (!$this->isAllowed($contribName)) {
+            return null;
+        }
+
+        // endpoint is only parsed if it is provided
+        $dsn = empty($endpointUrl) ? '' : DsnParser::parse($endpointUrl);
+        $endpointUrl = $this->parseBaseUrl($dsn);
+        // parameters are only retrieved if there was an endpoint given
+        $args = empty($dsn) ? [] : $dsn->getParameters();
+        
+        switch ($contribName) {
+            case 'jaeger':
+                return $exporter = $this->generateJaeger($endpointUrl);
+            case 'zipkin':
+                return $exporter = $this->generateZipkin($endpointUrl);
+            case 'newrelic':
+                return $exporter = $this->generateNewrelic($endpointUrl, $args['licenseKey'] ?? null);
+            case 'otlp':
+                return $exporter = $this->_generateOtlp();
+            case 'otlpgrpc':
+                return $exporter = $this->generateOtlpGrpc();
+            case 'zipkintonewrelic':
+                return $exporter = $this->generateZipkinToNewrelic($endpointUrl, $args['licenseKey'] ?? null);
+            }
+    }
+
+    private function isAllowed(string $exporter)
+    {
+        return array_key_exists($exporter, $this->allowedExporters) && $this->allowedExporters[$exporter];
+    }
+
+    // constructs the baseUrl from
+    private function parseBaseUrl($dsn)
+    {
+        if ($dsn == false) {
+            return null;
+        }
+        $parsedUrl = '';
+        $parsedUrl .= empty($dsn->getScheme()) ? '' : $dsn->getScheme() . '://';
+        $parsedUrl .= empty($dsn->getHost()) ? '' : $dsn->getHost();
+        $parsedUrl .= empty($dsn->getPort()) ? '' : ':' . $dsn->getPort();
+        $parsedUrl .= empty($dsn->getPath()) ? '' : $dsn->getPath();
+
+        return $parsedUrl;
+    }
+
+    private function generateJaeger(string $endpointUrl)
+    {
+        $exporter = new JaegerExporter(
+            $this->name,
+            $endpointUrl,
+            new Client(),
+            new HttpFactory(),
+            new HttpFactory()
+        );
+
+        return $exporter;
+    }
+    private function generateZipkin(string $endpointUrl)
+    {
+        $exporter = new ZipkinExporter(
+            $this->name,
+            $endpointUrl,
+            new Client(),
+            new HttpFactory(),
+            new HttpFactory()
+        );
+
+        return $exporter;
+    }
+    private function generateNewrelic(string $endpointUrl, $licenseKey)
+    {
+        if ($licenseKey == false) {
+            return null;
+        }
+        $exporter = new NewrelicExporter(
+            $this->name,
+            $endpointUrl,
+            $licenseKey,
+            new Client(),
+            new HttpFactory(),
+            new HttpFactory()
+        );
+
+        return $exporter;
+    }
+
+    private function _generateOtlp()
```

Comment:
might this underscore have been forgotten to be removed?

Reply:
yes. thank you good catch